lib/pbxProject.js

```diff
@@ -1265,15 +1284,16 @@ pbxProject.prototype.removeFromHeaderSearchPaths = function(file) {
 
     }
 }
-pbxProject.prototype.addToHeaderSearchPaths = function(file) {
+pbxProject.prototype.addToHeaderSearchPaths = function(file, opt) {
     var configurations = nonComments(this.pbxXCBuildConfigurationSection()),
         INHERITED = '"$(inherited)"',
         config, buildSettings, searchPaths;
 
     for (config in configurations) {
         buildSettings = configurations[config].buildSettings;
 
-        if (unquote(buildSettings['PRODUCT_NAME']) != this.productName)
+        var productName = opt && opt.productName ? opt.productName : this.productName;
+        if (unquote(buildSettings['PRODUCT_NAME']) != productName)
```

Comment:
Please us strict equal check

```suggestion
        if (unquote(buildSettings['PRODUCT_NAME']) !== productName)
```